src/result-list.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+var bind = require('lodash.bind');
+var ResponseHandler = require('./response-handler');
+
+bind.placeholder = '_';
+
+function ResultList(retriever, options) {
+  if (!retriever) {
+    throw new Error('Expected Retriever as an argument');
+  }
+
+  options || (options = {});
+
+  this._retriever = retriever;
+  this._onFailure = options.onFailure;
+  this._onSuccess = options.onSuccess;
+  this._page = options.page || 1;
+}
+
+ResultList.prototype.next = function() {
+  this._retriever.query({page: this._page});
+  this._page += 1;
+  return this._get(this._page - 1);
+};
+
+ResultList.prototype._get = function(page) {
+  return this._retriever
+    .get()
+    .then(bind(this._handleSuccess, this, '_', page))
+    .fail(bind(this._handleFailure, this, '_', page));
```

Comment:
Just switch the order of the arguments in `_handleSuccess` and `_handleFailure` and avoid the confusion of placeholders.